smart_open/tests/test_smart_open.py

```diff
@@ -191,6 +191,24 @@ def test_open_with_keywords_explicit_r(self):
             actual = fin.read()
         self.assertEqual(expected, actual)
 
+    @unittest.skipIf(
+        sys.version_info < (3, 4),
```

Comment:
better to check that `pathlib` can be imported or not (not python version, because I can have installed `pathlib` from backport.